backend-lib/src/main/rust/lib.rs

```diff
@@ -1560,24 +1566,64 @@ fn encode_transaction_data_request<'a>(
                 }
             };
 
+            let request_at = request_at.map_or(-1, |t| {
+                t.duration_since(UNIX_EPOCH)
+                    .expect("SystemTime should never be before the epoch")
+                    .as_secs()
+                    .try_into()
+                    .expect("we have time before a SystemTime overflows i64")
+            });
+
+            let tx_status_filter = env.new_object(
```

Comment:
The most likely thing to be failing in the PR is these two calls, which are attempting to construct Kotlin data objects from within JNI. I cannot find any documentation on how to do this, and am guessing that Kotlin implements them as empty-constructor Java classes.